src/deploy/firestore/prepare.ts

```diff
@@ -65,6 +80,30 @@ export default async function (context: any, options: any): Promise<void> {
     context.firestoreRules = true;
   }
 
-  prepareIndexes(context, options);
-  await prepareRules(context, options);
+  const firestoreConfigs: fsConfig.ParsedFirestoreConfig[] = fsConfig.getFirestoreConfig(
+    context.projectId,
+    options
+  );
+  if (!firestoreConfigs || firestoreConfigs.length === 0) {
+    return;
+  }
+
+  context.firestore = context.firestore || {};
+  context.firestore.indexes = [];
+  context.firestore.rules = [];
+  const rulesDeploy: RulesDeploy = new RulesDeploy(options, RulesetServiceType.CLOUD_FIRESTORE);
+  _.set(context, "firestore.rulesDeploy", rulesDeploy);
+
+  firestoreConfigs.forEach((firestoreConfig: fsConfig.ParsedFirestoreConfig) => {
```

Comment:
Normal for loops are strongly preferred over foreach, as they lead to less confusion about scope (go/tsjs-practices/iteration#avoid-using-foreach-when-other-approaches-are-better)